components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/DefaultOAuth2TokenValidator.java

```diff
@@ -110,6 +113,21 @@ public boolean validateScope(OAuth2TokenValidationMessageContext messageContext)
                             " are not found in the server configuration ", StringUtils.join(appScopeValidators, ", "),
                     app.getApplicationName(), OAuth2Util.getTenantDomainOfOauthApp(app)));
         }
+
+        // Deriving the global level scope validator implementations.
```

Comment:
Need to add the logic to handle allowed scopes

Reply:
Added in AccessTokenIssuer  in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1468/commits/328d264a2e70a4028ddfcbf05a25f57e3e873174